### PR TITLE
Another round of gRPC v2 api functions

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Adding more functions to gRPC v2 API.
Depends on https://github.com/Concordium/concordium-grpc-api/pull/14.

## Changes

- Extend gRPC api v2 with `GetCryptographicParameters`, `GetBlockInfo`, `GetBakerList`, `GetPoolStatus`, `GetPassiveDelegationStatus`, `GetBlocksAtHeight` and `GetTokenomicsStatus`.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-node/Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ codegen-units = 1
 [dependencies]
 # Internal dependencies
 crypto_common = { path = "../concordium-base/rust-src/crypto_common", features = ["encryption"] }
```
Comment (abizjak) [inline concordium-node/Cargo.toml:28]: Does id export common? If yes, then we can remove this dependency.

File: concordium-node/src/consensus_ffi/ffi.rs [hunk 2]
```diff
@@ -1423,6 +1611,172 @@ impl ConsensusContainer {
...
+                &mut out_data,
+                copy_to_vec_callback,
+            )
+        }
+        .try_into()?;
+        response.ensure_ok("block")?;
+        Ok((out_hash, out_data))
+    }
+
+    /// Get a stream of live blocks at a given height.
```
Comment (abizjak) [inline concordium-node/src/consensus_ffi/ffi.rs:1727]: Make sure to update documentation after changing this from a stream.

File: concordium-node/src/consensus_ffi/ffi.rs [hunk 3]
```diff
@@ -1584,6 +1938,15 @@ extern "C" fn copy_to_vec_callback(out: *mut Vec<u8>, data: *const u8, len: i64)
     out.extend_from_slice(data);
 }
 
+/// Copy cryptographic parameters to the target location provided by the
+/// `source` pointer.
+extern "C" fn copy_cryptographic_parameters_callback(
+    target: *mut Option<CryptographicParameters>,
+    source: *const CryptographicParameters,
+) {
+    unsafe { *target = source.as_ref().map(|g| g.to_owned()) };
```
Comment (abizjak) [inline concordium-node/src/consensus_ffi/ffi.rs:1947]: What about
```suggestion
    unsafe { *target = source.as_ref().cloned() };
```
?

**Review by abizjak — APPROVED** [review]
Looks good. Did you test and compare responses from the old and new APIs?

That should be easy for most of these queries.

File: concordium-consensus/src/Concordium/Queries.hs [hunk 4]
```diff
@@ -459,10 +491,10 @@ getBlockBirkParameters = liftSkovQueryBlock $ \bp -> do
 
 -- |Get the cryptographic parameters of the chain at a given block.
 -- The result is versioned (which will currently always be version 0).
```
Comment (Bargsteen) [inline concordium-consensus/src/Concordium/Queries.hs:493]: Outdated comment. 
What was the reasoning behind not returning `Versioned` anymore?

File: concordium-node/src/consensus_ffi/ffi.rs [hunk 5]
```diff
@@ -683,6 +710,129 @@ extern "C" {
...
+        block_id: *const u8,
+        out_hash: *mut u8,
+        out: *mut Vec<u8>,
+        copier: CopyToVecCallback,
+    ) -> i64;
+
+    /// Get a stream of live blocks at a given height.
+    ///
+    /// * `consensus` - Pointer to the current consensus.
+    /// * `stream` - Pointer to the response stream.
```
Comment (Bargsteen) [inline concordium-node/src/consensus_ffi/ffi.rs:797]: Remove if you turn it into a regular response.

File: concordium-node/src/consensus_ffi/ffi.rs [hunk 6]
```diff
@@ -683,6 +710,129 @@ extern "C" {
...
+    ///   otherwise relative.
+    /// * `genesis_index` - Genesis index to start from. Set to 0 to use
+    ///   absolute height.
+    /// * `restrict` - Whether to return results only from the specified genesis
+    ///   index (1), or allow results from more recent genesis indices as well
+    ///   (0).
+    /// * `callback` - Callback for writing to the response stream.
+    pub fn getBlocksAtHeightV2(
+        consensus: *mut consensus_runner,
+        stream: *mut futures::channel::mpsc::Sender<Result<Vec<u8>, tonic::Status>>,
```
Comment (Bargsteen) [inline concordium-node/src/consensus_ffi/ffi.rs:808]: Remove if you turn it into a regular response.

**Review by Bargsteen — APPROVED** [review]
LGTM 👍

File: concordium-node/Cargo.toml (hunk 1 above)
Comment (limemloh) [reply concordium-node/Cargo.toml:28]: It does not seem to.

File: concordium-consensus/src/Concordium/Queries.hs (hunk 4 above)
Comment (limemloh) [reply concordium-consensus/src/Concordium/Queries.hs:493]: There is only one version

**Pull request merged**